src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js

```diff
@@ -155,25 +178,78 @@ ApiGlobe.prototype.addImageryLayerFromJSON = function addImageryLayerFromJSON(ur
  * @return     {layer}  The Layers.
  */
 
-ApiGlobe.prototype.addImageryLayersFromJSONArray = function addImageryLayersFromJSONArray(urls) {
-    var proms = [];
+ApiGlobe.prototype.addImageryLayersFromJSONArray = function addImageryLayersFromJSONArray(urls, parentId) {
+    const proms = [];
 
-    for (var i = 0; i < urls.length; i++) {
-        proms.push(Fetcher.json(urls[i]).then(this.addImageryLayer.bind(this)));
+    for (const url of urls) {
+        proms.push(
+            Fetcher.json(url).then((layer) => {
+                layer.update = updateLayeredMaterialNodeImagery;
+                layer.graftedOnId = parentId;
```

Comment:
This code has changed. Now you can do:
```js
let scene = new Scene(...);
scene.addLayer(tile);
scene.addLayer(imagery, tile.id);
```
(see index_plane.html: https://github.com/iTowns/itowns2/pull/239/files#diff-912a8a195bb3f6b6ae7543308f9fe79fR96)